### PR TITLE
Add Docker image build & publish GHA

### PR DESCRIPTION
**Pull request by mindaugasrukas**
## What was changed
Added Docker image build & publish GHA

## Why?
Have pre-built images for testing.

## Checklist

1. Closes 
No
2. How was this tested:
GHA

3. Any docs updates needed?
No

### CODE REVIEW

File: .github/workflows/docker.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        run: echo '${{ toJSON(github) }}'
+
+      - name: Dump Input context
+        id: input_context_step
+        run: echo '${{ toJSON(inputs) }}'
+
+      - name: Checkout
+        uses: actions/checkout@v3
+      
+      - name: Set up QEMU
```
Comment (bergundy) [inline .github/workflows/docker.yaml:28]: Why is this needed?

File: .github/workflows/docker.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to DockerHub
```
Comment (bergundy) [inline .github/workflows/docker.yaml:34]: There’s a command that builds the image, this isn’t the way it’s supposed to be built

**Review by bergundy — COMMENTED** [review]
Let’s sync on what we want here, we had a different idea of how this image would be built

File: .github/workflows/docker.yaml (hunk 1 above)
Comment (mindaugasrukas) [reply .github/workflows/docker.yaml:28]: For having ARM images to run on macOS M1.

**Comment by Sushisource** [comment]
Closing in favor of https://github.com/temporalio/omes/pull/25
